src/zenml/mlstacks/utils.py

```diff
@@ -0,0 +1,525 @@
+#  Copyright (c) ZenML GmbH 2023. All Rights Reserved.
+
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+
+#       https://www.apache.org/licenses/LICENSE-2.0
+
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Functionality to handle interaction with the mlstacks package."""
+
+import json
+import uuid
+from functools import wraps
+from pathlib import Path
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Callable,
+    Dict,
+    List,
+    Optional,
+    Tuple,
+    Union,
+)
+
+import click
+import pkg_resources
+
+from zenml.cli.utils import (
+    error,
+    print_model_url,
+    verify_mlstacks_prerequisites_installation,
+)
+from zenml.client import Client
+from zenml.constants import (
+    MLSTACKS_SUPPORTED_STACK_COMPONENTS,
+    STACK_RECIPE_PACKAGE_NAME,
+)
+from zenml.enums import StackComponentType
+from zenml.utils.dashboard_utils import get_component_url, get_stack_url
+from zenml.utils.yaml_utils import read_yaml
+
+if TYPE_CHECKING:
+    from mlstacks.models import Component, Stack
+
+
+def verify_installation(func: Callable[..., Any]) -> Callable[..., Any]:
```

Comment:
Why does this need to exist, can't you just call `verify_mlstacks_prerequisites_installation()` inside the function instead? Both is one line but decorator is much harder to read.

Reply:
I took it out now and reverted back to your suggestion.